setup/config/config.go

```diff
@@ -293,19 +293,31 @@ func (config *Dendrite) Derive() error {
 }
 
 // SetDefaults sets default config values if they are not explicitly set.
-func (c *Dendrite) Defaults(generate bool) {
+func (c *Dendrite) Defaults(generate bool, isMonolith bool) {
```

Comment:
I'd prefer to see an opts struct like:
```go
type DefaultOpts {
    IsMonolith bool
    Generate bool
}
```
which is then used here. It makes it clearer at call-sites to know what `Defaults(true, true)` mean.